nodejs/tests/stream_reader.ts

```diff
@@ -0,0 +1,33 @@
+// Copyright Pravega Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { StreamManager } from '../stream_manager.js';
+import { StreamCut } from '../stream_reader_group.js';
+
+const stream_manager = StreamManager('tcp://127.0.0.1:9090', false, false, true);
+
+const stream_reader_group = stream_manager.create_reader_group(StreamCut.head(), 'rg1', 'scope1', 'stream1');
+const stream_reader = stream_reader_group.create_reader('r1');
+
+try {
+    const seg_slice = await stream_reader.get_segment_slice();
+    const dec = new TextDecoder('utf-8');
+    for (const event of seg_slice) {
+        const raw_value = event.data();
+        console.log(`Event at ${event.offset()} reads ${dec.decode(raw_value)}`);
+    }
+} catch (e) {
+    console.log(e);
```

Comment:
Is this going to work with things like github actions? We want to make sure that tests properly fail the build not just log the errors.

Reply:
No, it will not. Even tests got rewritten in #361, they are not meant to be run in github actions. If we need to run a standalone Pravega in actions, we should create a new issue.

The catch thing is to make sure other readers would be created correctly if this fails as we have a single standalone running when testing. But I add the random names so they should be removed. I want to do it in #361 since this file is removed entirely in the next pr.

Reply:
We definitely want to be able to have tests run as a pre-submit step. Otherwise they are likely to simply be not run. 
We currently separate unit tests and integration tests which run separately where unit tests are assumed to be parallelism safe and integration tests are not. If this is using the standalone, then it's definitely an integration test. So we should just make sure to include it in that pre-submit step and that any errors will cause it to fail.